subsys/net/openthread/Kconfig

```diff
@@ -19,12 +19,47 @@ config NORDIC_SECURITY_PROMPTLESS
 choice OPENTHREAD_SECURITY
 	depends on !OPENTHREAD_SECURITY_INTERNAL
 
+config OPENTHREAD_NRF_SECURITY_PSA_CHOICE
+	bool "nRF Security with PSA crypto enabled"
+	select OPENTHREAD_NRF_SECURITY_PSA
+
 config OPENTHREAD_NRF_SECURITY_CHOICE
 	bool "nRF Security"
 	select OPENTHREAD_NRF_SECURITY
 
 endchoice
 
+config OPENTHREAD_NRF_SECURITY_PSA
+	bool
+	select EXPERIMENTAL
+	select NRF_SECURITY
+	select MBEDTLS_ENABLE_HEAP
+	select MBEDTLS_PSA_CRYPTO_STORAGE_C if !BUILD_WITH_TFM
+	select PSA_NATIVE_ITS if !BUILD_WITH_TFM
+	select PSA_WANT_ALG_RSA_OAEP
```

Comment:
Does OpenThread use RSA? I noticed that flash increase can be reduced fro 31kB to 14kB if we do not enable RSA :)

Reply:
I cannot see the RSA define in OpenThread config: https://github.com/openthread/openthread/blob/main/third_party/mbedtls/mbedtls-config.h

@ArekBalysNordic do you think we need RSA support to be enabled here?

Reply:
No, we don't need it if OpenThread doesn't use it. Maybe it was a redundant config in my PR, I didn't optimize it recently, just ran it. :)

Reply:
I've removed this part and tested locally- commissioning/joining and COAPS look good. I'll schedule full nightly suite with PSA enabled too.